tls/s2n_quic_support.c

```diff
@@ -54,6 +54,11 @@ bool s2n_connection_is_quic_enabled(struct s2n_connection *conn)
     return (conn && conn->quic_enabled) || (conn && conn->config && conn->config->quic_enabled);
 }
 
+bool s2n_connection_is_client_resumption_enabled(struct s2n_connection *conn)
+{
+    return conn && conn->config && conn->config->use_tickets && conn->config->session_ticket_cb;
+}
```

Comment:
For naming, I think you should match the "[s2n_config_set_session_tickets_onoff](https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L1199)" method. So "s2n_connection_are_session_tickets_on" or "s2n_connection_are_session_tickets_enabled".

For the implementation, I'm still not sure about checking the callback. Technically you can use session tickets without the callback using the legacy s2n_connection_get_session method.

Reply:
I see what you mean, but you can't use session tickets in s2n-quic(or actually the rust bindings currently) without the callback. So if a user just enabled tickets but didn't actually set the callback, they wouldn't receive any tickets. This API is for quic, you can't call it without enabling quic, so I don't know why we wouldn't design it exclusively for quic.

Reply:
But you can call it without enabling quic-- nothing in the API itself requires quic to be enabled. It's just in the quic header.

If we can rely on the callback always being set in quic if session tickets are enabled on the config, why not just check the config?

Reply:
Okay, tbh I see your point and also I don't really have a strong opinion about how to do this. I'll change it to be only checking the config.